frontend/src/metabase/components/Header.jsx

```diff
@@ -39,104 +82,133 @@ export default class Header extends Component {
+                { isEditing && (
+                    <EditBar
+                        title={editingTitle}
+                        subtitle={editingSubtitle}
+                        // TODO - @kdoh - shouldnt bars have standard buttons?
+                        buttons={editBarButtons}
+                    />
+                )}
+                {
+                    // TODO - @kdoh - this is highly specific to the dashbaord
+                    // header and should be refactored out
+                }
+                <HeaderModal
+                    isOpen={!!this.props.headerModalMessage}
+                    height={this.state.headerHeight}
+                    title={this.props.headerModalMessage}
+                    onDone={this.props.onHeaderModalDone}
+                    onCancel={this.props.onHeaderModalCancel}
+                />
+                <div className="py1 lg-py2 xl-py3 wrapper flex align-center" ref="header">
                     <div className="Entity py3">
-                        {titleAndDescription}
-                        {attribution}
+                        {
+                            isEditingInfo
+                                ? (
+                                    <div className="Header-title flex flex-column flex-full bordered rounded my1">
+                                        <Input
+                                            className="AdminInput text-bold border-bottom rounded-top h3"
+                                            type="text"
+                                            value={item.name || ""}
+                                            onChange={this.setItemAttribute.bind(this, "name")}
+                                        />
+                                        <Input
+                                            className="AdminInput rounded-bottom h4"
+                                            type="text"
+                                            value={item.description || ""}
+                                            onChange={this.setItemAttribute.bind(this, "description")}
+                                            placeholder="No description yet"
+                                        />
+                                    </div>
+                                )
+                                : (
+                                    <TitleAndDescription
+                                        title={item && item.id
+                                                ? item.name
+                                                : `New ${objectType}`
+                                        }
+                                        description={item.description}
+                                    />
+                                )
+                        }
+                        {
+                            // TODO - @kdoh - does this ever evalutate to true?
```

Comment:
It does. `.Header-attribution` class has a `display: none` property so you will never see it 😆